src/io/flutter/logging/FlutterLogEntryParser.java

```diff
@@ -101,15 +102,29 @@ private FlutterLogEntry parseLoggingEvent(@NotNull Event event) {
       }
     }
 
-    // TODO(pq): If message.getValueAsStringIsTruncated() is true, we'll need to retrieve the full string
-    // value and update this entry after creation.
-    String messageStr = message.getValueAsString();
+    final String messageStr = message.getValueAsString();
+    final FlutterLogEntry entry = lineHandler.parseEntry(messageStr, category, level);
+
     if (message.getValueAsStringIsTruncated()) {
-      messageStr += "...";
-    }
+      entry.setMessage(messageStr + "...");
+      final Isolate isolate = new Isolate(json.get("isolate").getAsJsonObject());
+      debugProcess.getVmServiceWrapper().getObject(isolate.getId(), message.getId(), new GetObjectConsumer() {
+        @Override
+        public void received(Obj response) {
+          entry.setMessage(((Instance)response).getValueAsString());
+        }
 
-    final LineInfo lineInfo = parseLine(messageStr, category);
-    final FlutterLogEntry entry =  new FlutterLogEntry(timestamp(), lineInfo, level);
+        @Override
+        public void received(Sentinel response) {
+          // TODO(pq): handle?
```

Comment:
For this case and the error case I would just remove the ... and leave the original text perhaps adding <truncated> at the end of the text.